concent_api/gatekeeper/tests/test_gatekeeper_views.py

```diff
@@ -0,0 +1,141 @@
+            files                           = ['blender/benchmark/test_task/scene-Helicopter-27-cycles.blend', 'blender/benchmark/test_task/texture.tiff'],
+            operation                       = 'download'
+        )
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_accept_valid_message(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response, HttpResponse)
+        self.assertEqual(response.status_code, 200)
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_return_401_if_wrong_authorization_header(self):
+        golem_upload_token = dump(self.upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+        encrypted_token = b64encode(golem_upload_token).decode()
+        response1 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'GolemGolem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response1, JsonResponse)
+        self.assertEqual(response1.status_code, 401)
+        self.assertIn('message', response1.json().keys())
+
+        response2 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATIONN = 'Golem ' + encrypted_token,
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response2, JsonResponse)
+        self.assertEqual(response2.status_code, 401)
+        self.assertIn('message', response2.json().keys())
+
+        response3 = self.client.get('/gatekeeper/download-auth/blender/benchmark/test_task/scene-Helicopter-27-cycles.blend',
+            HTTP_AUTHORIZATION = 'Golem ' + 'encrypted_token',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertIsInstance(response3, JsonResponse)
+        self.assertEqual(response3.status_code, 401)
+        self.assertIn('message', response3.json().keys())
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_download_should_return_401_if_wrong_token_cluster_address(self):
+        upload_token = self.upload_token
+        upload_token.storage_cluster_address = 'www://storage.concent.golem.network/'
+        golem_upload_token = dump(upload_token, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
```

Comment:
You should not rely on setting values in tests. If you're using any settings, hard-code their values with `override_settings()`.

Reply:
Done. Added `@override_settings(
    CONCENT_PRIVATE_KEY = b'l\xcdh\x19\xeb$>\xbcG\xa1\xc7v\xe8\xd7o\x0c\xbf\x0e\x0fM\x89lw\x1e\xd7K\xd6Hnv$\xa2',
    CONCENT_PUBLIC_KEY  = b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;'
)` before each class.

Reply:
What about `STORAGE_CLUSTER_ADDRESS`? All the settings you rely on should be overridden. Not only those you refer to explicitly.